examples/sample_directory.rs

```diff
@@ -0,0 +1,324 @@
+    for (index, entry) in control.into_iter().enumerate() {
+        println!("...file {}/{}", index, control_len);
+        let original_size = std::fs::metadata(&entry.default_path).unwrap().len();
+        if original_size < min_size as u64 {
+            // Skip files that are too small.
+            continue;
+        }
+
+        let json = if let Ok(json) = parser.parse_file(&entry.default_path) {
+            json
+        } else {
+            // Could not parse source
+            continue;
+        };
+
+        let mut ast = if let Ok(ast) = binjs::specialized::es6::ast::Script::import(&json) {
+            ast
+        } else {
+            // Could not import AST.
+            continue;
+        };
+
+        println!("...compressing {:?}", entry.default_path);
+        println!(
+            "...original size size: {}",
+            original_size.separated_string()
+        );
+        total_original_size += original_size;
+
+        binjs::specialized::es6::scopes::AnnotationVisitor::new().annotate_script(&mut ast);
+
+        let destination_path = match dump_path {
+            Some(ref path) => Some(path.join(entry.relative_path.as_path())),
+            None => None,
+        };
+        let destination_path = match destination_path {
+            // Note: We need to keep `path` alive.
+            Some(ref path) => Some(path.as_ref()),
+            None => None,
+        };
+
+        let mut encoder = Encoder::new(destination_path, options.clone());
+
+        let mut serializer = binjs::specialized::es6::io::Serializer::new(encoder);
+        serializer
+            .serialize(&ast, &mut IOPath::new())
+            .expect("Could not generate dictionary");
+        let data = serializer.done().expect("Could not finalize dictionary");
+
+        let binjs_size = data.len() as u64;
+        println!("...binjs size: {}", binjs_size.separated_string());
+        total_binjs_size += binjs_size;
+
+        let brotli_size = {
+            let out = Command::new("brotli")
+                .arg("--best")
+                .arg("--keep")
+                .arg("--stdout")
+                .arg(&entry.default_path)
+                .output()
+                .expect("Error during brotli");
+            assert!(out.status.success());
+            assert!(out.stdout.len() != 0);
+            out.stdout.len()
+        } as u64;
+        println!("...brotli size: {}", brotli_size.separated_string());
+        total_brotli_size += brotli_size;
+
+        let binjs_div_brotli = binjs_size as f64 / brotli_size as f64;
+        println!("...binjs/brotli: {:.2}", binjs_div_brotli);
+
+        let compression = Compression {
+            original_size,
+            brotli_size,
+            binjs_size,
+            binjs_div_brotli,
+        };
+        results.push((compression, entry.default_path));
+    }
+
+    if show_results {
+        println!("** Ranked results (worse to best)");
+        results.sort_by(|a, b| {
```

Comment:
`.sort_by_key`?

Reply:
Can't `sort_by_key` with `f64`, unfortunately.

Reply:
Ah right. I've used https://github.com/reem/rust-ordered-float to work around that in the past, but maybe extra dependency is not worth it.